httpcore5/src/main/java/org/apache/hc/core5/http/ContentType.java

```diff
@@ -67,105 +68,125 @@ public final class ContentType implements Serializable {
     /**
      * Public constant media type for {@code application/pdf}.
      * @since 5.1
      */
     public static final ContentType APPLICATION_PDF = create(
-            "application/pdf", StandardCharsets.UTF_8);
+            "application/pdf", (Charset) null, true);
 
     public static final ContentType APPLICATION_SOAP_XML = create(
-            "application/soap+xml", StandardCharsets.UTF_8);
+            "application/soap+xml", StandardCharsets.UTF_8, false);
     public static final ContentType APPLICATION_SVG_XML = create(
-            "application/svg+xml", StandardCharsets.UTF_8);
+            "application/svg+xml", StandardCharsets.UTF_8, false);
     public static final ContentType APPLICATION_XHTML_XML = create(
-            "application/xhtml+xml", StandardCharsets.UTF_8);
+            "application/xhtml+xml", StandardCharsets.UTF_8, false);
     public static final ContentType APPLICATION_XML = create(
-            "application/xml", StandardCharsets.UTF_8);
+            "application/xml", StandardCharsets.UTF_8, false);
     /**
      * Public constant media type for {@code application/problem+json}.
      * @see <a href="https://tools.ietf.org/html/rfc7807#section-6.1">Problem Details for HTTP APIs, 6.1. application/problem+json</a>
      * @since 5.1
      */
     public static final ContentType APPLICATION_PROBLEM_JSON = create(
-            "application/problem+json", StandardCharsets.UTF_8);
+            "application/problem+json", StandardCharsets.UTF_8, true);
     /**
      * Public constant media type for {@code application/problem+xml}.
      * @see <a href="https://tools.ietf.org/html/rfc7807#section-6.2">Problem Details for HTTP APIs, 6.2. application/problem+xml</a>
      * @since 5.1
      */
     public static final ContentType APPLICATION_PROBLEM_XML = create(
-            "application/problem+xml", StandardCharsets.UTF_8);
+            "application/problem+xml", StandardCharsets.UTF_8, false);
 
     /**
      * Public constant media type for {@code application/rss+xml}.
      * @since 5.1
      */
     public static final ContentType APPLICATION_RSS_XML = create(
-            "application/rss+xml", StandardCharsets.UTF_8);
+            "application/rss+xml", StandardCharsets.UTF_8, false);
 
     public static final ContentType IMAGE_BMP = create(
-            "image/bmp");
+            "image/bmp", (Charset) null, true);
     public static final ContentType IMAGE_GIF = create(
-            "image/gif");
+            "image/gif", (Charset) null, true);
     public static final ContentType IMAGE_JPEG = create(
-            "image/jpeg");
+            "image/jpeg", (Charset) null, true);
     public static final ContentType IMAGE_PNG = create(
-            "image/png");
+            "image/png", (Charset) null, true);
     public static final ContentType IMAGE_SVG = create(
-            "image/svg+xml");
+            "image/svg+xml", (Charset) null, false);
```

Comment:
Well, it is still XML, means text...

Reply:
HI @michael-o 
 I understand that SVG is XML-based, and UTF-8 is commonly used as the default encoding for XML documents. However, leaving the charset as null in this case allows flexibility for other charsets to be specified in the SVG document itself, if needed (e.g., ISO-8859-1). This ensures the ContentType class respects the encoding declared within the SVG document rather than forcing UTF-8 by default. If you believe UTF-8 should be enforced due to its predominance in XML use cases, I can make the change

Reply:
The same argument would apply to any XML type document: leave charset out.

Reply:
cahnged